src/Error/ExceptionRenderer.php

```diff
@@ -90,6 +95,26 @@ class ExceptionRenderer implements ExceptionRendererInterface
      */
     protected $request;
 
+    /**
+     * Map of exceptions to http status codes.
+     *
+     * This can be customized for users that don't want specific exceptions to throw 404 errors
+     * or want their application exceptions to be automatically converted.
+     *
+     * @var array
+     * @psalm-var array<string, int>
+     */
+    protected $exceptionHttpCodes = [
+        // Controller exceptions
+        MissingActionException::class => 404,
+        // Datasource exceptions
+        InvalidPrimaryKeyException::class => 404,
```

Comment:
I'm thinking if `InvalidPrimaryKeyException` should be treated as developer error and return 500.

It is about correct number of passed keys in `$table->get(null)`, `$table->get(1)`, `$table->get([1, 2])`, not value itself, it is responsibility of developer to ensure it is valid size of array, so it is not client side error.

Parameter is always null/string in default baked controller, array can't be passed from URL.

Reply:
@garas That is not always true in all cases I think
If it comes from e.g. query string, a user or malicious person could most likely also input an array into what goes into such a method.
Depends on how the code has been written.

Reply:
I too am leaning towards `InvalidPrimaryKeyException` being a `500` instead of `404`. It's the devs job to satanize the value coming from the request.

Reply:
I can get behind changing InvalidPrimaryKeyException to a 500 as well. It is likely to be an uncommon occurrence, both in the number of effected applications and ways to trigger this exception.

Reply:
Changed InvalidPrimaryKeyException to render with 500 code since it seems to be a strong opinion here.